server/aesdsocket.c

```diff
@@ -0,0 +1,411 @@
+				//}
+	//		}
+	//		syslog(LOG_DEBUG, "recv success\n");
+	//	}
+	//}
+		
+	//if(!fork())
+	//{
+	//	close(sockfd); //child doesn't need listener
+	//	status = send(fd, buf, RX_MAX_LEN, RECV_FLAGS);
+	//	if(status == -1)
+	//	{
+	//		syslog(LOG_ERR, "send error = %d\n",errno);
+	//		fprintf(stderr, "send error: %d\n", errno);
+	//		continue;
+	//	}
+	//	else
+	//	{
+	//		syslog(LOG_DEBUG, "send success\n");
+	//		close(fd);
+	//	}
+	//}
+	//close(fd);
+	// above commented is the one I had tried reference: Beej's guide
+	// seeked help from Dhiraj for daemon process part
+    	if(d_arg)
+	{
+		pid = fork();
+		if(pid == -1)
+		{
+			syslog(LOG_ERR, "fork error = %d\n",errno);
+			fprintf(stderr, "fork error: %d\n", errno);
+			return -1;
+		}
+		else if(pid != 0)
+		{
+			exit(0);
+		}
+		
+		status = setsid();
+		if(status == -1)
+		{
+			syslog(LOG_ERR, "setsid error = %d\n",errno);
+			fprintf(stderr, "setsid error: %d\n", errno);
+			return -1;
+		}
+		else
+		{
+			syslog(LOG_DEBUG, "setsid success\n");
+		}
+		
+		status = chdir("/");
+		if(status == -1)
+		{
+			syslog(LOG_ERR, "chdir error = %d\n",errno);
+			fprintf(stderr, "chdir error: %d\n", errno);
+			return -1;
+		}
+		
+		open("/dev/null", O_RDWR);
+		dup(0);
+		dup(0);
+	}
+
+    	status = listen(serverfd, MAX_CONNECTIONS_ON_INCOMING_QUEUE);
+	if(status == -1)
+	{
+		syslog(LOG_ERR, "listen error = %d\n",errno);
+		fprintf(stderr, "listen error: %d\n", errno);
+		//exit(-1);
+	}
+	else
+	{
+		syslog(LOG_DEBUG, "listen success\n");
+	}
+	
+	//sa.sa_handler = sigchld_handler; // reap all dead processes
+	//sigemptyset(&sa.sa_mask);
+	//sa.sa_flags = SA_RESTART;
+	
+	//status = sigaction(SIGCHLD, &sa, NULL);
+	//if(status == -1)
+	//{
+	//	syslog(LOG_ERR, "sigaction error = %d\n",errno);
+	//	fprintf(stderr, "sigaction error: %d\n", errno);
+	//	exit(-1);
+	//}
+	//else
+	//{
+	//	syslog(LOG_DEBUG, "sigaction success\n");
+	//}
+	
+	//printf("server: waiting for connections...\n");
+
+    filefd = open(FILE_PATH, O_RDWR | O_CREAT | O_TRUNC, S_IRWXU | S_IRWXG | S_IRWXO);
+
+    while(1)
+    {
+        addr_size = sizeof(c);
+        acceptfd = accept(serverfd, (struct sockaddr_in *) &c, &addr_size);
+	if(acceptfd == -1)
+	{
+		syslog(LOG_ERR, "accept error = %d\n",errno);
+		fprintf(stderr, "accept error: %d\n", errno);
+		exit(-1);
+	}
+	else
+	{
+		syslog(LOG_DEBUG, "accept success\n");
+		syslog(LOG_DEBUG, "Accepted connection from : %s\n", inet_ntoa(c.sin_addr));
+	}
+	
+        max_buf_size = MAX_BUF_SIZE;
+        saved_bytes = 0;
+
+
+        write_packet = malloc(sizeof(char) * MAX_BUF_SIZE);
+
+        bool packet_collected = false;
+        while(!packet_collected)
+        {
+            recv_bytes = recv(acceptfd, server_buf, MAX_BUF_SIZE, RECV_FLAGS);
+
+            if (recv_bytes == 0 || (strchr(server_buf, '\n') != NULL))
+            {
+                packet_collected = true;
+            }
+
+            if ((max_buf_size - saved_bytes) < recv_bytes)
```

Comment:
@abaa3023 I think when this check fails you will have alredy exceeded array bounds on server_buf?  You need to ensure you have at least `MAX_BUF_SIZE` bytes available before calling `recv`.